tests/rpminspect/rpminspect.sh

```diff
@@ -50,45 +49,72 @@ rlJournalStart
 		fi
-		# Run rpminspect for the specified architectures
-		[[ -n "$RPMINPSECT_ARCHES" ]] && args="$args --arches=$RPMINPSECT_ARCHES"
-		# If we have a previous build to compare with, use that as before_build
-		[[ -n "$latest_build" ]] && args="$args ./inspect_builds/$latest_build"
-		# The remaining part is treated as the after_build/the build to be inspected
-		args="$args ./inspect_builds/$PACKIT_PACKAGE_NVR"
+		[[ -n "$RPMINPSECT_ARCHES" ]] && rlRun "args=\"\$args --arches=$RPMINPSECT_ARCHES\"" 0 "Set args: Run rpminspect for the specified architectures"
+		[[ -n "$latest_build" ]] && rlRun "args=\"\$args ./inspect_builds/$latest_build\"" 0 "Set args: Use latest_build as before_build"
+		rlRun "args=\"\$args --threshold=${RPMINSPECT_THRESHOLD:-BAD}\"" 0 "Set args: Failure threshold"
+		[[ -n "${RPMINSPECT_SUPPRESS}" ]] && rlRun "args=\"\$args --suppress=${RPMINSPECT_SUPPRESS}\"" 0 "Set args: Suppress threshold"
+		rlRun "args=\"\$args ./inspect_builds/$PACKIT_PACKAGE_NVR\" && echo \$args" 0 "Set args: Set downloaded source as after_build"
 	  rlRun "/usr/bin/rpminspect $args" 0 "Run rpminspect"
+		# Get the number of BAD and VERIFY check results
+		# Note about jq syntax used:
+		# - .[][]               : collapse the dict of list of objects into a series of objects
+		# - select(.result ...) : filter the series of objects by the `result` value
+		# - [ ... ] | length    : create a list of the filtered object series and count the length
+		rlRun "bad_results=\$(jq '[.[][] | select(.result == \"BAD\")] | length' $TMT_TEST_DATA/result.json)" 0 "Count the BAD check results"
+		rlRun "verify_results=\$(jq '[.[][] | select(.result == \"VERIFY\")] | length' $TMT_TEST_DATA/result.json)" 0 "Count the VERIFY check results"
+		rlRun "info_results=\$(jq '[.[][] | select(.result == \"INFO\")] | length' $TMT_TEST_DATA/result.json)" 0 "Count the INFO check results"
+		rlRun "ok_results=\$(jq '[.[][] | select(.result == \"OK\")] | length' $TMT_TEST_DATA/result.json)" 0 "Count the OK check results"
 		rlRun "cp $TMT_PLAN_DATA/viewer.html $TMT_TEST_DATA/viewer.html"
 	rlPhaseEnd
 
 	rlPhaseStartCleanup
 		rlRun "popd"
 		rlRun "rm -r $tmp" 0 "Remove tmp directory"
-		# Note: $TESTRESULT_RESULT_STRING is not made available. Using $__INTERNAL_PHASES_WORST_RESULT instead
-		cat <<EOF > "$TMT_TEST_DATA/results.yaml"
-- name: /rpminspect
-  result: ${__INTERNAL_PHASES_WORST_RESULT,,}
+	rlPhaseEnd
+rlJournalEnd
+
+# Report the test results
+source $BEAKERLIB_DIR/TestResults
+if [[ ${TESTRESULT_RESULT_STRING,,} != pass ]]; then
+	# If test failed or triggerred warning within the test itself, use that outcome
+	result=${TESTRESULT_RESULT_STRING,,}
+else
+	# Otherwise take the highest result status of the rpminspect checks
+	# BAD > VERIFY > INFO > OK
+	if (( bad_results > 0 )); then
+		result=fail
+	elif (( verify_results > 0 )); then
+		# TODO: Switch result to warn when testing-farm supports it
```

Comment:
Do we have a ticket to watch?

Reply:
It seems like https://gitlab.com/testing-farm/oculus/-/issues/21. But it should also be handled by the packit GH Application so that it displays as `notification` instead/as well as a `pass` status. (Funny how GH allows that for GH applications, but not for actions :face_exhaling:)